src/Utils/QueryAnalyzer.php

```diff
@@ -483,46 +512,42 @@ public function track_nodes( $model ) {
 		return $model;
 	}
 
-
 	/**
-	 * Return headers
-	 *
-	 * @param array $headers The array of headers being returned
+	 * Returns graphql keys for use in debugging and headers.
 	 *
 	 * @return array
 	 */
-	public function get_headers( array $headers = [] ): array {
+	public function get_graphql_keys() {
 
-		$keys = [];
+		if ( ! empty( $this->graphql_keys ) ) {
+			return $this->graphql_keys;
+		}
+
+		$keys        = [];
+		$return_keys = '';
 
 		if ( $this->get_query_id() ) {
 			$keys[] = $this->get_query_id();
 		}
 
 		if ( ! empty( $this->get_root_operation() ) ) {
-			$headers['X-GraphQL-Operation-Type'] = $this->get_root_operation();
-			$keys[]                              = 'graphql:' . $this->get_root_operation();
+			$keys[] = 'graphql:' . $this->get_root_operation();
 		}
 
 		if ( ! empty( $this->get_list_types() ) && is_array( $this->get_list_types() ) ) {
-			$headers['X-GraphQL-List-Types'] = implode( ' ', array_unique( array_values( $this->get_list_types() ) ) );
-			$keys                            = array_merge( $keys, $this->get_list_types() );
-
+			$keys = array_merge( $keys, $this->get_list_types() );
 		}
 
 		if ( ! empty( $this->get_runtime_nodes() ) && is_array( $this->get_runtime_nodes() ) ) {
-			$headers['X-GraphQL-Nodes'] = implode( ' ', array_unique( array_values( $this->get_runtime_nodes() ) ) );
```

Comment:
same here

Reply:
This is the one that really seems to build up the most and is already output in the `X-GraphQL-Keys` header, so removed this to reduce the overall header size. It was output separately for convenience over necessity.